###**Choosing Dataset**

###Project 2 - Part 1 (Core)

In [ ]:
# imports
import pandas as pd


In [25]:
### LOAD DATASET FROM GOOGLE DRIVE OR URL
# load data
path_file = '/content/drive/MyDrive/StudentFolders16wk20wk/CodingDojo/02-MachineLearning/Week08/Data/wines_SPA.csv'
df_wine = pd.read_csv(path_file)
df_wine

winery             wine  year  rating  num_reviews  \
0            Teso La Monja            Tinto  2013     4.9           58   
1                   Artadi    Vina El Pison  2018     4.9           31   
2             Vega Sicilia            Unico  2009     4.8         1793   
3             Vega Sicilia            Unico  1999     4.8         1705   
4             Vega Sicilia            Unico  1996     4.8         1309   
...                    ...              ...   ...     ...          ...   
7495               Contino          Reserva  2016     4.2          392   
7496  Conreria d'Scala Dei    Les Brugueres  2018     4.2          390   
7497           Mustiguillo  Finca Terrerazo  2017     4.2          390   
7498           Matarromera     Gran Reserva  2011     4.2          389   
7499              Sei Solo         Preludio  2016     4.2          388   

     country            region   price                  type  body  acidity  
0     Espana              Toro  995.00              Toro Red   5.0      3.0  
1     Espana    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2     Espana  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3     Espana  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4     Espana  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0  
...      ...               ...     ...                   ...   ...      ...  
7495  Espana             Rioja   19.98             Rioja Red   4.0      3.0  
7496  Espana          Priorato   16.76           Priorat Red   4.0      3.0  
7497  Espana      El Terrerazo   24.45                   Red   4.0      3.0  
7498  Espana  Ribera del Duero   64.50  Ribera Del Duero Red   5.0      3.0  
7499  Espana  Ribera del Duero   31.63  Ribera Del Duero Red   5.0      3.0  

[7500 rows x 11 columns]

In [26]:
df_wine.head()

winery           wine  year  rating  num_reviews country  \
0  Teso La Monja          Tinto  2013     4.9           58  Espana   
1         Artadi  Vina El Pison  2018     4.9           31  Espana   
2   Vega Sicilia          Unico  2009     4.8         1793  Espana   
3   Vega Sicilia          Unico  1999     4.8         1705  Espana   
4   Vega Sicilia          Unico  1996     4.8         1309  Espana   

             region   price                  type  body  acidity  
0              Toro  995.00              Toro Red   5.0      3.0  
1    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0

In [27]:
df_wine.isnull().sum()

winery            0
wine              0
year              2
rating            0
num_reviews       0
country           0
region            0
price             0
type            545
body           1169
acidity        1169
dtype: int64

### 1. Source of the data


https://www.kaggle.com/datasets/fedesoriano/spanish-wine-quality-dataset

### 2. Brief description of the data

This dataset is related to red variants of spanish wines. The dataset describes several popularity and description metrics their effect on it's quality.
The task is to predict either the quality of wine or the prices using the given data.

### 3. What is the target?

Quality of wine

### 4. What does one row represent? (A person? A business? An event? A product?)

A product which is a bottle of wine.

### 5. Is this a classification or regression problem?

Classification

### 6. How many features does the data have?

12

### 7. How many rows are in the dataset?

In [28]:
df_wine.shape

(7500, 11)

7500 rows

### 8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

1. Missing Data
2. Imbalanced class
3. Data encoding
4. Outliers in numerical columns
5. Feature Engineering

###Dataset 2

###Stroke Prediction Dataset

In [30]:
### LOAD DATASET FROM GOOGLE DRIVE OR URL
# load data
path_file = '/content/drive/MyDrive/StudentFolders16wk20wk/CodingDojo/02-MachineLearning/Week08/Data/healthcare-dataset-stroke-data.csv'
df_stroke = pd.read_csv(path_file)
df_stroke

id  gender   age  hypertension  heart_disease ever_married  \
0      9046    Male  67.0             0              1          Yes   
1     51676  Female  61.0             0              0          Yes   
2     31112    Male  80.0             0              1          Yes   
3     60182  Female  49.0             0              0          Yes   
4      1665  Female  79.0             1              0          Yes   
...     ...     ...   ...           ...            ...          ...   
5105  18234  Female  80.0             1              0          Yes   
5106  44873  Female  81.0             0              0          Yes   
5107  19723  Female  35.0             0              0          Yes   
5108  37544    Male  51.0             0              0          Yes   
5109  44679  Female  44.0             0              0          Yes   

          work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0           Private          Urban             228.69  36.6  formerly smoked   
1     Self-employed          Rural             202.21   NaN     never smoked   
2           Private          Rural             105.92  32.5     never smoked   
3           Private          Urban             171.23  34.4           smokes   
4     Self-employed          Rural             174.12  24.0     never smoked   
...             ...            ...                ...   ...              ...   
5105        Private          Urban              83.75   NaN     never smoked   
5106  Self-employed          Urban             125.20  40.0     never smoked   
5107  Self-employed          Rural              82.99  30.6     never smoked   
5108        Private          Rural             166.29  25.6  formerly smoked   
5109       Govt_job          Urban              85.28  26.2          Unknown   

      stroke  
0          1  
1          1  
2          1  
3          1  
4          1  
...      ...  
5105       0  
5106       0  
5107       0  
5108       0  
5109       0  

[5110 rows x 12 columns]

In [31]:
df_stroke.isnull().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64

### 1. Source of the data


https://www.kaggle.com/datasets/fedesoriano/heart-failure-prediction

### 2. Brief description of the data

According to the World Health Organization (WHO) stroke is the 2nd leading cause of death globally, responsible for approximately 11% of total deaths.
This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status. Each row in the data provides relavant information about the patient.

### 3. What is the target?

Stroke

###4. What does one row represent? (A person? A business? An event? A product?)

An event

###5. Is this a classification or regression problem?

Classification

###6. How many features does the data have?

12 Features

###7. How many rows are in the dataset?

5110 rows

In [33]:
df_stroke.shape

(5110, 12)